src/main/java/net/minecraftforge/fml/common/Loader.java

```diff
@@ -680,78 +682,110 @@ public ModClassLoader getModClassLoader()
-            }
-            // after elements are things that load before we do (so they are out dependencies)
-            else if ("required-after".equals(instruction) || "after".equals(instruction))
-            {
-                dependencies.add(VersionParser.parseVersionReference(target));
-            }
-            else
-            {
-                parseFailure = true;
-            }
-        }
+                if (hasSide)
+                {
+                    // If this is a required element, add it to the required list
+                    if (!hasRequired && "required".equals(instruction))
+                    {
+                        if (!targetIsAll)
+                        {
+                            requirements.add(VersionParser.parseVersionReference(target));
+                            hasRequired = true;
+                        }
+                        else
+                        {
+                            parseFailure(dep, "You can't require everything");
+                        }
+                    }
+                    else if (hasRequired && "required".equals(instruction))
+                    {
+                        parseFailure(dep, "Required or other instructions after Required have already been defined");
+                    }
+                    else
+                    {
+                        hasRequired = true;
+                    }
 
-        if (parseFailure)
-        {
-            FMLLog.log(Level.WARN, "Unable to parse dependency string %s", dependencyString);
-            throw new LoaderException(String.format("Unable to parse dependency string %s", dependencyString));
+                    if (hasRequired)
+                    {
+                        if (!hasOrder)
+                        {
+                            // before elements are things we are loaded before (so they are our dependants)
+                            if ("before".equals(instruction))
+                            {
+                                dependants.add(VersionParser.parseVersionReference(target));
+                                hasOrder = true;
+                            }
+                            // after elements are things that load before we do (so they are out dependencies)
+                            else if ("after".equals(instruction))
+                            {
+                                dependencies.add(VersionParser.parseVersionReference(target));
+                                hasOrder = true;
+                            }
+                        }
+                        else
+                        {
+                            parseFailure(dep, "Order instruction has already been defined");
+                        }
+                    }
+                }
+            }
         }
     }
+    
+    private static void parseFailure(String dependencyString, String cause){
```

Comment:
What field? This is a method instead of a field. Its name should be a verb, meaning an action.